utils/neuralmagic/utils.py

```diff
@@ -113,6 +114,78 @@ def nm_log_console(message: str, logger: "Logger" = None, level: str = "info"):
             logger.info(f"{colorstr('Neural Magic: ')}{message}")
 
 
+def neuralmagic_onnx_export(
+    model: torch.nn.Module,
+    sample_data: torch.Tensor,
+    weights_path: Path,
+    one_shot: Optional[str],
+    dynamic: Optional[Dict[str, int]],
+    output_names: List[str],
+) -> Path:
+    """
+    Augmented ONNX export to optimize and properly post-process sparsified models
+
+    :param model: model to export
+    :param sample_data: data to be used with export
+    :weights path: path from which the torch model was loaded. Used only for save
+        pathing and naming purposes
+    :one_shot: one_shot recipe, if one was applied. Used only for save pathing and
+        naming purposes
+    :param dynamic: dictionary of input or output names to list of dimensions
+        of those tensors that should be exported as dynamic
+    :output_names: names of output tensors
+
+    :return: path to saved ONNX model
+    """
+
+    # If the target model is a SparseZoo or YOLOv5 Hub model, save it to a
+    # DeepSparse_Deployment directory at the working directory root. Inside, create a
+    # subdirectory based on model/stub name
+    if str(weights_path).startswith("zoo:") or not len(weights_path.parents):
+        sub_dir = (
+            str(weights_path).split("zoo:")[1].replace("/", "_")
+            if str(weights_path).startswith("zoo:")
+            else weights_path
+        )
+
+        # If one-shot applying a recipe, update name to convey the starting stub/model
+        # and the one_shot recipe applied
+        if one_shot:
+            sub_dir = f"{sub_dir}_one_shot_{one_shot}"
```

Comment:
Won't the folder name get too big if the entire stub is used?

Reply:
Unfortunately yeah. The second example in the PR description is about as long as the folder name will get, which is pretty long. Seems like an ok price to pay for a quick and general solution, but am open to being convinced otherwise. 

Also, a good compromise could be to always omit the cv_detection part since that's going to be common to all stubs used here